src/api/c/convolve.cpp

```diff
@@ -113,9 +115,20 @@ af_err convolve2_sep(af_array *out, af_array col_filter, af_array row_filter, co
 
         af_dtype signalType  = sInfo.getType();
 
-        dim4 signalDims = sInfo.dims();
+        dim4 sdims = sInfo.dims();
+
+        ARG_ASSERT(1, (sdims.ndims()>=2));
+
+        if (cfInfo.isScalar() && rfInfo.isScalar()) {
+            af_array colArray, rowArray, filter;
+
+            AF_CHECK(af_tile(&colArray, col_filter, sdims[0], sdims[1], sdims[2], sdims[3]));
+            AF_CHECK(af_tile(&rowArray, row_filter, sdims[0], sdims[1], sdims[2], sdims[3]));
+            AF_CHECK(af_mul (&filter, colArray, rowArray, false));
+
+            return af_mul(out, signal, filter, false);
+        }
```

Comment:
But may be this particular case is something like a once in a blue moon case for separable convolution. I am not saying we shouldn't do it, but just that this can be much more tersely handled by the user if one checks if filter is scalar before calling separable convolution.